include/experimental/__p2630_bits/submdspan_extents.hpp

```diff
@@ -0,0 +1,238 @@
+namespace std {
+namespace experimental {
+
+// Slice Specifier allowing for strides and compile time extent
+template <class OffsetType, class ExtentType, class StrideType>
+struct strided_index_range {
+  using offset_type = OffsetType;
+  using extent_type = ExtentType;
+  using stride_type = StrideType;
+
+  OffsetType offset;
+  ExtentType extent;
+  StrideType stride;
+};
+
+namespace detail {
+
+// Mapping from submapping ranks to srcmapping ranks
+// InvMapRank is gonna be a index_sequence, which we build recursively
+// to contain the mapped indicies.
+template <size_t Counter, class InvMapRank, class... SliceSpecifiers>
+struct inv_map_rank;
+
+// specialization reducing rank by one (i.e. integral slice specifier)
+template <size_t Counter, class Slice, class... SliceSpecifiers,
+          size_t... MapIdxs>
+requires(is_convertible_v<Slice, size_t>) struct inv_map_rank<
+    Counter, index_sequence<MapIdxs...>, Slice, SliceSpecifiers...> {
+  using type = typename inv_map_rank<Counter + 1, index_sequence<MapIdxs...>,
+                                     SliceSpecifiers...>::type;
+};
+
+// specialization for slice specifiers expressing some form of range
+template <size_t Counter, class Slice, class... SliceSpecifiers,
+          size_t... MapIdxs>
+requires(!is_convertible_v<Slice, size_t>) struct inv_map_rank<
+    Counter, index_sequence<MapIdxs...>, Slice, SliceSpecifiers...> {
+  using type =
+      typename inv_map_rank<Counter + 1, index_sequence<MapIdxs..., Counter>,
+                            SliceSpecifiers...>::type;
+};
+
+// final specialziation containing the final index_sequence
```

Comment:
A minor spelling issue:
```suggestion
// final specialization containing the final index_sequence
```
Does "final specialization" mean "this is the end of the 'recursion' on `inv_map_rank` specializations"?

Reply:
yeah

Reply:
I updated the comment to express this better